src/input/mod.rs

```diff
@@ -2241,7 +2251,9 @@ impl State {
                 }
             }
             if let Some(or) = shell.override_redirect_windows.iter().find(|or| {
-                or.is_in_input_region(&(global_pos.as_logical() - or.geometry().loc.to_f64()))
+                or.is_in_input_region(
+                    &(global_pos.as_logical() - X11Surface::geometry(*or).loc.to_f64()),
```

Comment:
Another issue caused by the wrong `geometry` method being called? And I guess the fact `or` is a double reference makes it not use the implementation one would expect. Smithay probably needs more coordinate space types, and maybe fewer overloaded names.

Reply:
Yep. Only stepping through it I noticed the problem..

Reply:
Actually it's not a `&&X11Surface` here.